server/BudgetTracker.Business/Api/BudgetApi.cs

```diff
@@ -153,5 +149,38 @@ public async Task<ApiResponse> GetRootBudgets(ApiRequest request)
             response = new ApiResponse(responseData);
             return response;
         }
-    }    
+
+        public Task<ApiResponse> FetchBudgetTree(ApiRequest request)
+        {
+            User user = await Authenticate(request);
+            ApiResponse response;
+
+            Guid rootBudgetId = request.Arguments<FetchBudgetTreeArgumentsApiContract>();
+
+            try
+            {
+                Budget rootBudget = GetBudgetOrErrorIfUnauthorized(rootBudgetId, user.Id);
+
+                await _budgetRepository.LoadBudgetTree(rootBudget);
+            }
+            catch (RepositoryException ex)
+            {
+                response.Error = ex.Message;
+            }
+
+            return response;
+        }
+
+        private async Task<Budget> GetBudgetOrErrorIfUnauthorized(Guid budgetId, Guid userId)
```

Comment:
Like here you could just add a summary tag that states it will return and error on failure.